NewsFeed/views/CCBRNewsViewController.m

```diff
@@ -109,26 +111,33 @@ - (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSe
     return self.viewModel.itemCount;
 }
 
+- (void)collectionView:(UICollectionView *)collectionView willDisplayCell:(UICollectionViewCell *)cell forItemAtIndexPath:(NSIndexPath *)indexPath {
+    BOOL shouldLoadMore = indexPath.row >= self.viewModel.itemCount - 3;
+    if (shouldLoadMore) {
+        [self.viewModel loadMore];
+    }
```

Comment:
Can you please move this logic to the view-model?
The **view-controller** should reflect the state in the **view-model**, it should not "know" the logic or "decide" when to load more.
If you put the logic of "loading more" in the **view-controller**, it will be hard to unit-test logic. Because you will need to load the view, fill the data and wait for the `willDisplayCell` method to be called!

Reply:
Also can you please optimize the `collection view` when loading more cell?

Reply:
- I did move the loading more condition to view models
- Optimize collection view when appending new data by inserting only new rows instead of reload the whole collection view.